src/app/filelogger.cpp

```diff
@@ -0,0 +1,267 @@
+}
+
+void FileLogger::setEnabled(bool value)
+{
+    settings()->storeValue(KEY_ENABLED, value);
+}
+
+QString FileLogger::path()
+{
+    return settings()->loadValue(KEY_PATH, QVariant(Utils::Fs::QDesktopServicesDataLocation() + LOG_FOLDER)).toString();
+}
+
+void FileLogger::setPath(const QString &value)
+{
+    settings()->storeValue(KEY_PATH, value);
+}
+
+bool FileLogger::isBackup()
+{
+    return settings()->loadValue(KEY_BACKUP, true).toBool();
+}
+
+void FileLogger::setBackup(bool value)
+{
+    settings()->storeValue(KEY_BACKUP, value);
+}
+
+bool FileLogger::isDeleteOld()
+{
+    return settings()->loadValue(KEY_DELETEOLD, true).toBool();
+}
+
+void FileLogger::setDeleteOld(bool value)
+{
+    settings()->storeValue(KEY_DELETEOLD, value);
+}
+
+int FileLogger::maxSize()
+{
+    int val = settings()->loadValue(KEY_MAXSIZE, 10).toInt();
+    if (val < 1)
+        return 1;
+    if (val > 1000)
+        return 1000;
+    return val;
+}
+
+void FileLogger::setMaxSize(const int &value)
+{
+    settings()->storeValue(KEY_MAXSIZE, std::min(std::max(value, 1), 1000));
+}
+
+int FileLogger::age()
+{
+    int val = settings()->loadValue(KEY_AGE, 6).toInt();
+    if (val < 1)
+        return 1;
+    if (val > 365)
+        return 365;
+    return val;
+}
+
+void FileLogger::setAge(const int &value)
+{
+    settings()->storeValue(KEY_AGE, std::min(std::max(value, 1), 365));
+}
+
+int FileLogger::ageType()
+{
+    int val = settings()->loadValue(KEY_AGETYPE, 1).toInt();
+    return (val < 0 || val > 2) ? 1 : val;
+}
+
+void FileLogger::setAgeType(const int &value)
+{
+    settings()->storeValue(KEY_AGETYPE, (value < 0 || value > 2) ? 1 : value);
+}
+
+void FileLogger::configure()
+{
+    QString tmpPath = Utils::Fs::fromNativePath(path());
+    if (!tmpPath.endsWith('/'))
+        tmpPath += "/";
+    QDir dir(tmpPath);
+    dir.mkpath(tmpPath);
+    tmpPath += "qbittorrent.log";
+
+    if (tmpPath != m_path) {
+        m_path = tmpPath;
+
+        if (m_logFile) {
+            closeLogFile();
+            delete m_logFile;
+        }
+        m_logFile = new QFile(m_path);
+        openLogFile();
+    }
+
+    if (isDeleteOld()) {
+        QDateTime date = QDateTime::currentDateTime();
+
+        switch (static_cast<FileLogAgeType>(ageType())) {
+        case DAYS:
+            date = date.addDays(age());
+            break;
+        case MONTHS:
+            date = date.addMonths(age());
+            break;
+        default:
+            date = date.addYears(age());
+        }
+
+        foreach (const QFileInfo file, dir.entryInfoList(QStringList() << "qbittorrent.log.bak*", QDir::Files | QDir::Writable, QDir::Time | QDir::Reversed)) {
+            if (file.lastModified() < date)
+                break;
+            Utils::Fs::forceRemove(file.absoluteFilePath());
+        }
+    }
+}
+
+void FileLogger::addLogMessage(const Log::Msg &msg)
+{
+    if (!m_logFile) return;
+
+    QTextStream str(m_logFile);
+    QString type;
```

Comment:
The variable may be removed, perhaps. `str << "(I) "` does not look worse than  `type = "(I) "`